### PR TITLE
Login, Register Page UI and Functionality and Firebase Authentication

### PR DESCRIPTION
**Pull request by thisaiveerapandian**
Added Login, Register Page UI and Functionality, and Firebase Authentication Integration.

### CODE REVIEW

**Comment by ehsankkk1** [comment]
Hi pandian , great work but could you please make smaller commits since 800 line in one commit is a little to much

File: android/app/build.gradle [hunk 1]
```diff
@@ -44,7 +45,7 @@ android {
 
     defaultConfig {
         // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
-        applicationId "org.helpapaw.helpapaw"
```
Comment (BurningAXE) [inline android/app/build.gradle:47]: Why did you change the app Id?

File: lib/src/controllers/user_controller.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import '../services/constants.dart' as global;
+
+class UserController {
+  late SharedPreferences _prefs;
+
+  Future<bool> get isUserLoggedIn async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
```
Comment (BurningAXE) [inline lib/src/controllers/user_controller.dart:8]: This is temporary, I guess? :)

File: lib/src/models/users.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Users {
```
Comment (BurningAXE) [inline lib/src/models/users.dart]: it's best to have models in singular - i.e. `User`

File: lib/src/pages/splash.dart [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  Widget build(BuildContext context) {
+    SystemChrome.setPreferredOrientations([
+      DeviceOrientation.portraitUp,
+    ]);
+
+    Future.delayed(const Duration(seconds: 2), () async {
+      bool isLoggedIn = await UserController().isUserLoggedIn;
+      Navigator.of(context).push(
+        MaterialPageRoute(
+            builder: (context) => isLoggedIn ? const HomePage(title: 'Help a Paw') : const LoginPage(),
```
Comment (BurningAXE) [inline lib/src/pages/splash.dart:21]: Actually we don't want to prevent users from viewing signals when not logged in. It's required only for active operations (e.g. submit a signal, comment, etc)

File: lib/src/services/constants.dart [hunk 5]
```diff
@@ -0,0 +1,10 @@
+// ROUTES
+const String homeRoute = '/home';
```
Comment (BurningAXE) [inline lib/src/services/constants.dart:2]: Good thing constants are extracted👍

File: android/app/build.gradle (hunk 1 above)
Comment (thisaiveerapandian) [reply android/app/build.gradle:47]: firebase config will automatically create an application id with 'com' that's y I changed the application id.

File: lib/src/controllers/user_controller.dart (hunk 2 above)
Comment (thisaiveerapandian) [reply lib/src/controllers/user_controller.dart:8]: yes. future we can use direct firebase login validation.

File: lib/src/models/users.dart (hunk 3 above)
Comment (thisaiveerapandian) [reply lib/src/models/users.dart]: ok. i will change

File: lib/src/pages/splash.dart (hunk 4 above)
Comment (thisaiveerapandian) [reply lib/src/pages/splash.dart:21]: ok. so before submitting the signal we have to show the alert please log in and then proceed right?

File: lib/src/services/constants.dart (hunk 5 above)
Comment (thisaiveerapandian) [reply lib/src/services/constants.dart:2]: ok i will change

File: lib/src/pages/splash.dart (hunk 4 above)
Comment (BurningAXE) [reply lib/src/pages/splash.dart:21]: Yes, exactly.

File: android/app/build.gradle (hunk 1 above)
Comment (BurningAXE) [reply android/app/build.gradle:47]: I changed the configuration to use the original app Ids. So, it stays the same after firebase config.

File: lib/src/pages/splash.dart (hunk 4 above)
Comment (thisaiveerapandian) [reply lib/src/pages/splash.dart:21]: Ok i will change

File: lib/src/models/users.dart (hunk 3 above)
Comment (thisaiveerapandian) [reply lib/src/models/users.dart]: done.

**Review by BurningAXE — APPROVED** [review]
